src/core/lib/promise/party.h

```diff
@@ -130,7 +129,10 @@ class PartySyncUsingAtomics {
         if (poll_one_participant(i)) {
           const uint64_t allocated_bit = (1u << i << kAllocatedShift);
           prev_state &= ~allocated_bit;
-          state_.fetch_and(~allocated_bit, std::memory_order_release);
+          uint64_t finished_prev_state =
```

Comment:
The changes here and on line 213 look unrelated to the rest of this PR.  Am I missing something?

Reply:
yeah it's a bit unrelated... worth doing though... I'll update the description.